doc/metadata_routing.rst

```diff
@@ -0,0 +1,153 @@
+
+.. _metadata_routing:
+
+Metadata Routing
+================
+
+This guide demonstrates how metadata such as ``sample_weight`` can be routed
```

Comment:
Maybe "sample-wise auxiliary data" or "sample-wise companion data" would be more precise?

Otherwise I am also fine to keep calling it metadata to for the sake of simplicity. No strong opinion.

Reply:
Although it is the consensus term that we reached, I'm not very happy with the use of "metadata" either, since that tends to mean a description of the data or its provenance. I'm not sure about calling them data at all. "auxiliary variables"? "Auxiliary information"? "Auxiliary vectors"? "Auxiliary arrays"?

Reply:
"auxiliary variables" is nice in the sense that we can implicitly guess that it's a set of sample-aligned datastructures (as are input and target variables for instance).